### PR TITLE
Updated Data Management Page

### PR DESCRIPTION
**Pull request by howard-e**
This PR is based on work done in #688.

This updates the Test Management Page to the Data Management Page proposed in #648, and implements the following P0 requirements:

**Test Plan**
- [x] Show the name of the test plan

**Covered AT**
- [x] As a near term stopgap, List covered AT. Works for now since we only have 3.

**Overall Status**
* Show the most advanced phase this plan has completed using the following logic:
	* If there is a recommended version:
		- [x] Show string "Recommended Since DATE" where DATE is the date on which a version of this plan was first recommended. Note that multiple versions may have progressed to recommended, so this is the date on which the earliest version progressed to recommended.
		- [x] If there are also versions currently active in any of the candidate, draft, or R&D complete phases, append string "+N New Versions in Progress" where N is the number of phases where a version is active. If N=1, use singular "version".
	- else if there is a candidate version:
		- [x] Show string "Candidate Review Started DATE" where DATE is the date on which the candidate review phase began. Note that multiple versions may have progressed to candidate, so this is the date on which the earliest version progressed to candidate.
		- [x] If there are also versions currently active in either the draft or R&D complete phases, append string "+N New Versions in Progress" where N is the number of phases where a version is active. If N=1, use singular "version".
	- else if there is a draft version:
		- [x] Show string "Draft Review Started DATE" where DATE is the date on which the draft review phase began. Note that multiple versions may have progressed to draft, so this is the date on which the earliest version progressed to draft.
		- [x] If there is also a version currently active in the draft phase, append string "+1 New Version in Progress".
	- else if R&D is complete:
		- [x] Show string "R&D Complete on DATE" where DATE is the date from the version identifier of the version that is currently ready for draft review.

**R&D Version**
- [x] If the latest version of the plan is in the draft, candidate, or recommended phase, show string "N/A".
* Else If the latest version has completed R&D and has not yet been advanced to draft review, show:
	- [x] Link with text "VERSION_STRING" that targets the single-page view of the plan. Example: [V22.12.08](https://aria-at.netlify.app/review/alert.html)
	- [x] If current user is an admin, show "Advance to Draft" button, but note that:
		- [x] If there is an earlier version that is draft and that version has some test plan runs in the test queue, this button will run the process for updating existing reports and preserving data for tests that have not changed.
		- [x] If there is an earlier version in the draft phase, this button will sunset that version. This will also sunset any reports completed using that version.

**Draft Review**
- [x] If a version of the plan is not in the draft phase and there are no versions in later phases, show string "Not Started"
- [x] If a version of the plan is not in the draft phase and there is a version in at least one of candidate or recommended phases, show string "Review of VERSION_STRING completed DATE", where:
	- VERSION_STRING identifies the version that most recently completed draft review.
	- DATE is the date that the version identified by VERSION_STRING completed draft review and advanced to candidate review.
	- Example: "Review of V22.12.08 completed Jan 30, 2023"
- If there is a version currently in the draft phase, show:
	- [x] Link with text "VERSION_STRING" that targets the single-page view of the plan. Example: [V22.12.08](https://aria-at.netlify.app/review/alert.html)
	- [x] If required reports are complete and user is an admin, show "Advance to Candidate" button. Note that:
		- [x] If there is an earlier version that is candidate and that version has some test plan runs in the test queue, this button will run the process for updating existing reports and preserving data for tests that have not changed.
		- [x] If there is an earlier version in the candidate phase, this button will sunset that version. This will also sunset any reports completed using that version.

**Candidate Review**
- [x] If a version of the plan is not in the candidate phase and there has not yet been a recommended version, show string "Not Started"
- [x] If a version of the plan is not in the candidate phase and there is a recommended version, show string "Review of VERSION_STRING completed DATE" where:
	- VERSION_STRING identifies the version that most recently completed candidate review.
	- DATE is the date that the version identified by VERSION_STRING completed draft review and advanced to candidate review.
	- Example: "Review of V22.12.08 completed Jan 30, 2023"
- If there is a version currently in the candidate phase, show:
	- [x] Link with text "VERSION_STRING" that targets the single-page view of the plan. Example: [V22.12.08](https://aria-at.netlify.app/review/alert.html)
	- [x] Show string "N Open Review Issues" and if N>=2, append " from N AT" Examples: "3 Open Review Issues from 2 AT" or "0 Open Review Issues"
	- [x] Show button "Advance to Recommended" when the following conditions are met:
		- [x] If there has not yet been a recommended version and open issues = 0 and days in review > 120 and user is admin, show the button.
		- [x] If there is already a recommended version and open review issues = 0 and user is admin, show the button.
		- [x] If there is an earlier version that is recommended and that version has some test plan runs in the test queue, this button will run the process for updating existing reports and preserving data for tests that have not changed.
		- [x] if there is an earlier version in the recommended phase, this button will sunset that version. This will also sunset any reports completed using that version.

**Recommended Version**
- [x] If a version of the plan is not in the recommended phase, shows the string "None Yet"
- If there is a version currently in the recommended phase, show:
	- [x] Link with text "VERSION_STRING" that targets the single-page view of the plan. Example: [V22.12.08](https://aria-at.netlify.app/review/alert.html)
	- [x] Show string "Approved DATE" where DATE is the date that the currently recommended version was advanced to the recommended phase.

**Confirmation prompts for advancing a version from one phase to next**
- [x] When a button is activated for advancing a test plan version to the next phase, a confirmation prompt is provided. The prompt must include the name of the test plan, its version, the covered AT, and the name of the new phase.

~**Note on why I'm drafting this PR for now. The links provided with the "VERSION_STRING" text has been left unchecked even though it works for _some_ links. That's because the link may not exist for some test plan versions because the state of the `aria-at/tests` folder at that point in time would not have had the built files being expected, so a more robust solution is required.**~

___

_This page can also be viewed at https://aria-at-app-sandbox.bocoup.com/data-management_

### CODE REVIEW

**Comment by howard-e** [comment]
Undrafting due to https://github.com/w3c/aria-at-app/pull/713/commits/281c18bea7a9c4b7be2f7303c7ab3c475ec7a7dd which addresses the (now striked out) note in the top comment

File: client/components/DataManagement/DataManagementRow/index.jsx
Comment (alflennik) [inline client/components/DataManagement/DataManagementRow/index.jsx:245]: I was playing around with the data management, and I was trying to figure out what would happen when we introduce a new draft version when there is already a recommended version. What happened was, the draft version never appeared in the data management page. Is that intended?

FYI the test plan I was looking at was the Disclosure Navigation Menu Example.

**Review by alflennik — COMMENTED** [review]
While I did have one big question so far, which related to how multiple versions are supposed to be shown simultaneously, I can see the seeds here for a much more powerful data system, and it's super cool to imagine what has just become possible with this new UI. I think it's really well thought out especially given how tricky the underlying version management problem is. It looks great.

File: client/components/DataManagement/DataManagementRow/index.jsx
Comment (howard-e) [reply client/components/DataManagement/DataManagementRow/index.jsx:469]: Thanks for that note!

File: client/components/DataManagement/DataManagementRow/index.jsx
Comment (howard-e) [reply client/components/DataManagement/DataManagementRow/index.jsx:939]: This is an implementation detail described in #648 for the Recommended Version Details Column:
`If a version of the plan is not in the recommended phase, show the string "None Yet"`.

Seems like the context being the finality of the "version" makes it sound appropriate? Like "We have no recommended versions yet" vs "we have not started a recommended version", which indicates that this version would also go through a review process, which wouldn't be true. But maybe it's something to discuss further?

File: client/components/DataManagement/DataManagementRow/index.jsx
Comment (howard-e) [reply client/components/DataManagement/DataManagementRow/index.jsx:245]: You were right that there was an issue here! There was a missing check to verify that there is a newer version for the test plan to be shown in the row, if at a lower phase. https://github.com/w3c/aria-at-app/pull/713/commits/a4b2585475974f20fb10d955e33169103761a7dc should address this.

File: client/components/DataManagement/DataManagementRow/index.jsx
Comment (alflennik) [reply client/components/DataManagement/DataManagementRow/index.jsx:939]: Ah ok, I guess it's cool if it's specifically designed that way.

**Comment by alflennik** [comment]
@howard-e Once the tests which are making a ton of warnings is fixed, I'm good to approve. Fantastic work with a very complex system.

**Comment by howard-e** [comment]
> @howard-e Once the tests which are making a ton of warnings is fixed, I'm good to approve. Fantastic work with a very complex system.

@alflennik thanks for catching that, https://github.com/w3c/aria-at-app/pull/713/commits/6d187976ae51d08ecf76d1edbe1048ac4e87f46d should address it

**Comment by alflennik** [comment]
I verified that the tests are no longer emitting warnings. Great work on this feature!

FYI I noticed when testing that there was a step missing in the database docs so I snuck it in to this branch.

**Review by alflennik — APPROVED** [review]

**Comment by mcking65** [comment]
@alflennik wrote:

> While I did have one big question so far, which related to how multiple versions are supposed to be shown simultaneously, I can see the seeds here for a much more powerful data system, and it's super cool to imagine what has just become possible with this new UI. I think it's really well thought out especially given how tricky the underlying version management problem is. It looks great.

Yay, this is an awesome vote of confidence!! Thank you @alflennik.

**Comment by howard-e** [comment]
This needs a rebase but will wait until #722 can be approved and merged into here as it's a P0 item, to avoid additional conflicts updating this may cause for that PR.

**Pull request merged**
